Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -302,6 +317,35 @@ - (BOOL)implicitMentionsEnabled {
     return [self.delegate implicitSearchLength] > 0;
 }
 
++ (nonnull NSCharacterSet *)whitespaceSet {
+    static NSCharacterSet *whitespaceSet;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        whitespaceSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+    });
+    return whitespaceSet;
+}
+
++ (nonnull NSCharacterSet *)punctuationSet {
+    static NSCharacterSet *punctuationSet;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        punctuationSet = [NSCharacterSet punctuationCharacterSet];
+    });
+    return punctuationSet;
+}
+
++ (nonnull NSCharacterSet *)separatorSet {
```

Comment:
please add comments to all these

and ideally we should migrate to only use this one right? or is there any case where we need to differentiate between whitespace and punctuation?

Reply:
Added comments. Yes, we do have cases where we need differentiation between white space and punctuation like in `characterTyped` method.